cli/ops/fs.rs

```diff
@@ -397,10 +397,22 @@ fn op_remove(
     let metadata = std::fs::symlink_metadata(&path)?;
     debug!("op_remove {} {}", path.display(), recursive);
     let file_type = metadata.file_type();
-    if file_type.is_file() || file_type.is_symlink() {
+    if file_type.is_file() {
       std::fs::remove_file(&path)?;
     } else if recursive {
       std::fs::remove_dir_all(&path)?;
+    } else if file_type.is_symlink() {
+      #[cfg(unix)]
+      std::fs::remove_file(&path)?;
+      #[cfg(not(unix))]
+      {
+        let is_symlink_dir = std::fs::metadata(&path)?.is_dir();
```

Comment:
Don't call fs::metadata() again.
It's already available to in 'file_type'.

Reply:
thanks, @piscisaureus 👍
now I'm using winapi to realize that symlink target type is a dir or file.

Reply:
I appreciate your effort, but this is not quite going in the right direction.

The metadata for the symlink is read literally just a few lines back: https://github.com/denoland/deno/pull/5488/files#diff-c7f29ec99d5cc9bbb158a5af5942c39dR400.

Calling std::fs::metafata() on a symlink is unreliable -- it'd fail if the target doesn't exist. You need to use the symlink Metadata from above.

Also, it would be great to add a test so we can see that it works.

Reply:
I'm sorry - I see that you actually did what I asked you to do.

Reply:
A test would still be very much appreciated though.

Reply:
actually, we have a test to remove symlink already [here](https://github.com/denoland/deno/blob/c4fe58d8df51e8f07142803ec0522f862fd30a3f/std/fs/ensure_symlink_test.ts#L134).
BUT, we didn't implement symlink creation in windows yet  [[i'm working on here](https://github.com/denoland/deno/compare/master...disizali:implement_symlink_windows?expand=1)], so we can't run remove tests without Deno.symlink implemented.

> also, another thing, [implement symlink for windows](https://github.com/denoland/deno/compare/master...disizali:implement_symlink_windows?expand=1), needs to this PR landed to run tests that I mentioned before, successfully ...

Reply:
> BUT, we didn't implement symlink creation in windows yet

Maybe do this for the time being?

```js
let s1 = await Deno.run({ cmd: ['cmd', '/c', 'mklink', 'file_link', 'foo'] }).status();
let s2 = await Deno.run({ cmd: ['cmd', '/c', 'mklink', '/d', 'dir_link', 'bar'] }).status();
assert(s1.success && s2.success);
```

> [i'm working on here], so we can't run remove tests without Deno.symlink implemented.

Awesome!

Reply:
Thanks, done.